### PR TITLE
fix: Fix echobot to compile with strict warnings.

### PR DESCRIPTION
**Pull request by iphydf**
This change is [Reviewable](https://reviewable.io/reviews/toktok/toxins/29)

### CODE REVIEW

File: echobot/echobot.c [hunk 1]
```diff
@@ -40,9 +40,13 @@ Tox *create_tox()
 
         uint8_t *savedata = malloc(fsize);
 
-        fread(savedata, fsize, 1, f);
+        size_t read_size = fread(savedata, fsize, 1, f);
         fclose(f);
 
+        if (read_size != fsize) {
```
Comment (robinlinden) [inline echobot/echobot.c:46]: `fread` returns the number of objects read, so this should be compared with `1`, not `fsize`?

**Review by robinlinden — CHANGES_REQUESTED** [review]

File: echobot/echobot.c [hunk 2]
```diff
@@ -40,9 +40,13 @@ Tox *create_tox()
 
         uint8_t *savedata = malloc(fsize);
```
Comment (robinlinden) [inline echobot/echobot.c:41]: Since you're adding error checks, should this malloc also be guarded?

File: echobot/echobot.c (hunk 1 above)
Comment (iphydf) [reply echobot/echobot.c:46]: Done.

File: echobot/echobot.c (hunk 2 above)
Comment (iphydf) [reply echobot/echobot.c:41]: Done.

**Review by robinlinden — APPROVED** [review]

**Pull request merged**